src/runtime/storageNG/backing-storage-proxy.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Copyright (c) 2017 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {CRDTTypeRecord} from '../crdt/crdt.js';
+import {StorageCommunicationEndpointProvider, ProxyMessage, ProxyCallback, StorageCommunicationEndpoint} from './store.js';
+import {PropagatedException} from '../arc-exceptions.js';
+import {ChannelConstructor} from '../channel-constructor.js';
+import {StorageProxy} from './storage-proxy.js';
+import {Dictionary} from '../hot.js';
+import {Handle} from './handle.js';
+import {Type} from '../type.js';
+import {assert} from '../../platform/assert-web.js';
+
+export class BackingStorageProxy<T extends CRDTTypeRecord> implements StorageCommunicationEndpointProvider<CRDTTypeRecord> {
+  private storageProxies: Dictionary<StorageProxy<CRDTTypeRecord>> = {};
+  private callbacks: Dictionary<ProxyCallback<CRDTTypeRecord>> = {};
+  private storageEndpoint: StorageCommunicationEndpoint<T>;
+  private storageKey: string;
+  private type: Type;
+
+  constructor(storeProvider: StorageCommunicationEndpointProvider<T>, type: Type, storageKey: string) {
+    this.storageEndpoint = storeProvider.getStorageEndpoint(this);
+    this.storageEndpoint.setCallback(msg => this.onMessage(msg));
```

Comment:
fyi my comment above about passing the function directly instead of wrapping in a lambda does have a caveat, and it's this case: if it's a member function (`this.onMessage`) then to ensure the `this` pointer is correct inside that member function, you either use a lambda as you have here, or explicitly bind it: `setCallback(this.onMessage.bind(this))`. Yay, JS.